src/lib/config.ts

```diff
@@ -342,6 +366,39 @@ export const LOGS_SUBSCRIPTION_EXPIRY_HEARTBEAT_SECONDS =
 
 export const LOGS_DEFAULT_RETENTION_LIMIT = 1000;
 
+export const LOGS_PRIMARY_BACKEND = optionalVar(
+	'LOGS_PRIMARY_BACKEND',
+	'redis',
+);
+if (LOGS_PRIMARY_BACKEND !== 'redis' && LOGS_PRIMARY_BACKEND !== 'loki') {
+	throw new Error('LOGS_PRIMARY_BACKEND must be either "redis" or "loki"');
+}
+if (LOGS_PRIMARY_BACKEND === 'loki' && !LOGS_LOKI_ENABLED) {
+	throw new Error(
+		'LOGS_PRIMARY_BACKEND cannot be "loki" if LOGS_LOKI_ENABLED is false',
+	);
+} else if (LOGS_PRIMARY_BACKEND === 'redis' && !LOGS_REDIS_ENABLED) {
+	throw new Error(
+		'LOGS_PRIMARY_BACKEND cannot be "redis" if LOGS_REDIS_ENABLED is false',
+	);
+}
+switch (LOGS_PRIMARY_BACKEND) {
```

Comment:
Is there any reason for swich case in here? Imo it is just more readable to continue `if/else` logic flow as above 

```
if (LOGS_PRIMARY_BACKEND === 'redis') {
...
} else if (LOGS_PRIMARY_BACKEND === 'loki') {
 ...
}

Reply:
It mostly comes from typescript complaining if I used `!== 'redis' elseif !== 'loki'` but it's ok with `===`